### PR TITLE
Remove COURSE_CATALOG_API_URL

### PR DESCRIPTION
**Pull request by vkaracic**
Follow-up to the TODO left in `ecommerce/core/models.py`.

@edx/helio please review.

---
TODO:
- [ ] Remove the `use_multi_tenant_discovery_api_urls` switch
- [x] Update docker devstack to include `discovery-api-url` flag value when running create_or_update_site (It's already [set](https://github.com/edx/devstack/blob/master/provision-ecommerce.sh#L7))

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @vkaracic! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

<a   href="https://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx%2Fecommerce&number=1427"   rel="nofollow" >Create an OSPR issue for this pull request</a>.

File: ecommerce/core/models.py [hunk 1]
```diff
@@ -381,10 +380,7 @@ def discovery_api_client(self):
             EdxRestApiClient: The client to access the Discovery service.
         """
 
-        # TODO Once the change is verified remove the switch and COURSE_CATALOG_API_URL from settings.
-        if waffle.switch_is_active('use_multi_tenant_discovery_api_urls') and self.discovery_api_url:
-            return EdxRestApiClient(self.discovery_api_url, jwt=self.access_token)
-        return EdxRestApiClient(settings.COURSE_CATALOG_API_URL, jwt=self.access_token)
+        return EdxRestApiClient(self.discovery_api_url, jwt=self.access_token)
```
Comment (iivic) [inline ecommerce/core/models.py:390]: What happens when the discovery_api_url is not set?

**Review by iivic — COMMENTED** [review]
You should verify with someone if this feature is approved just to make sure nothing breaks.

File: ecommerce/core/models.py (hunk 1 above)
Comment (vkaracic) [reply ecommerce/core/models.py:390]: Now it will be a required field, as it should be since I the ecommerce service is dependent on the discovery service.

**Comment by vkaracic** [comment]
@edx/helio please review.

**Review by iivic — APPROVED** [review]

File: ecommerce/core/management/commands/create_or_update_site.py [hunk 2]
```diff
@@ -128,7 +128,7 @@ def add_arguments(self, parser):
                             action='store',
                             dest='discovery_api_url',
                             type=str,
-                            required=False,
+                            required=True,
                             default='',
```
Comment (clintonb) [inline ecommerce/core/management/commands/create_or_update_site.py]: Remove this default. Also, set blank to False.

Comment (clintonb) [reply ecommerce/core/management/commands/create_or_update_site.py]: Ignore the bit about blank.

File: ecommerce/core/tests/test_commands.py [hunk 3]
```diff
@@ -184,7 +184,25 @@ def test_update_site(self):
          '--theme-scss-path=site/sass/css/', '--payment-processors=cybersource'],
         ['--site-id=1', '--site-domain=fake.server', '--partner-name=fake_partner',
          '--theme-scss-path=site/sass/css/', '--payment-processors=cybersource',
-         '--segment-key=abc']
+         '--segment-key=abc'],
+        ['--site-id=1', '--site-domain=fake.server', '--partner-name=fake_partner',
```
Comment (clintonb) [inline ecommerce/core/tests/test_commands.py:188]: Why did you add so many cases?

Comment (vkaracic) [reply ecommerce/core/tests/test_commands.py:188]: Because when new required arguments were added, this test was not updated to include them. I'm including the `discovery_api_url` argument together with all the other omitted ones.

**Review by clintonb — APPROVED** [review]

**Comment by clintonb** [comment]
Squash your commits and link to an appropriate ticket.
